test/common/wpt.js

```
@@ -6,9 +6,92 @@ const fs = require('fs');
 const fsPromises = fs.promises;
 const path = require('path');
 const events = require('events');
+const os = require('os');
 const { inspect } = require('util');
 const { Worker } = require('worker_threads');
 
+function getBrowserProperties() {
+  const release = !!process.release.sourceUrl;
+  const browser = {
+    browser_channel: release ? 'stable' : 'experimental',
+    browser_version: process.version,
+  };
+
+  if (release) {
+    browser.browser_version = process.version;
+  } else if (process.env.NODE_REVISION) {
+    browser.browser_version = process.version.replace('pre', process.env.NODE_REVISION);
+  }
+
+  return browser;
+}
+
+function getOs() {
+  switch (os.type()) {
+    case 'Linux':
+      return 'linux';
+    case 'Darwin':
+      return 'mac';
+    case 'Windows_NT':
+      return 'win';
+    default:
+      return 'unknown';
```

should we throw an error on this case?